design/sketch.webidl

```diff
@@ -5,6 +5,17 @@ typedef long u32;
 // ****************************************************************************
 
 // Buffer
+
+callback WebGPUMappedMemorySuccessCallback = void (ArrayBuffer);
+callback WebGPUMappedMemoryErrorCallback = void (DOMString);
+interface WebGPUMappedMemory {
+    bool isPending();
+    ArrayBuffer getPointer();
+
+    void then(WebGPUMappedMemorySuccessCallback success,
```

Comment:
I can't find any references to a standard, but I believe `then` method of a thenable should return a new `Promise`.

Reply:
On this point, if the caller only uses the pollable interface, could we confirm the creation of a `Promise` is avoided entirely?

Reply:
@dmikis I don't know for `then`-ables, but it looks to be the case for `Promises` (I'm sure you know already ^^) https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then

@grovesNL correct, the Promise wouldn't get created. The only overhead would be some unused members of `WebGPUMappedMemory`.

Reply:
@dmikis While I don't think a Thenable strictly speaking has to return anything in particular, you're right that it would be useful if it did.